packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -418,33 +423,44 @@ export class SummaryManager extends EventEmitter implements IDisposable {
             return this.nextSummarizerP;
         }
 
-        const loader = this.context.loader;
-
-        // TODO eventually we may wish to spawn an execution context from which to run this
-        const request: IRequest = {
-            headers: {
-                [LoaderHeader.cache]: false,
-                [LoaderHeader.clientDetails]: {
-                    capabilities: { interactive: false },
-                    type: summarizerClientType,
+        let summarizer: ISummarizer | undefined;
+        // back-compat 0.33: this function is added in 0.34, but the original path
+        // needs to be preserved until 0.33 is no longer supported at this layer
+        if (!this.context.createNextSummarizerFn) {
+            const loader = this.context.loader;
+
+            // TODO eventually we may wish to spawn an execution context from which to run this
+            const request: IRequest = {
+                headers: {
+                    [LoaderHeader.cache]: false,
+                    [LoaderHeader.clientDetails]: {
+                        capabilities: { interactive: false },
+                        type: summarizerClientType,
+                    },
+                    [DriverHeader.summarizingClient]: true,
+                    [LoaderHeader.reconnect]: false,
+                    [LoaderHeader.sequenceNumber]: this.context.deltaManager.lastSequenceNumber,
+                    [LoaderHeader.executionContext]: this.enableWorker ? "worker" : undefined,
                 },
-                [DriverHeader.summarizingClient]: true,
-                [LoaderHeader.reconnect]: false,
-                [LoaderHeader.sequenceNumber]: this.context.deltaManager.lastSequenceNumber,
-                [LoaderHeader.executionContext]: this.enableWorker ? "worker" : undefined,
-            },
-            url: "/_summarizer",
-        };
-
-        const response = await loader.request(request);
-
-        if (response.status !== 200
-            || (response.mimeType !== "fluid/object" && response.mimeType !== "fluid/component")) {
-            return Promise.reject(new Error("Invalid summarizer route"));
-        }
+                url: "/_summarizer",
+            };
+
+            const response = await loader.request(request);
 
-        const rawFluidObject = response.value as IFluidObject;
-        const summarizer = rawFluidObject.ISummarizer;
+            if (response.status !== 200
+                || (response.mimeType !== "fluid/object" && response.mimeType !== "fluid/component")) {
+                return Promise.reject(new Error("Invalid summarizer route"));
+            }
+
+            const rawFluidObject = response.value as IFluidObject;
+            summarizer = rawFluidObject.ISummarizer;
+        } else {
+            const nextSummarizer = await this.context.createNextSummarizerFn(
+                this.context.deltaManager.lastSequenceNumber,
+                this.enableWorker ? "worker" : undefined,
```

Comment:
Can we also extract the enableWorker flag all the way up to the Container?  Aside from simplifying the function signature would be good to consolidate all the opinionation on "how" to create a summarizer to one place I think.

Reply:
Newer iteration removes it from the function signature and keeps it in the headers.  For the future I think it can be added to the `IContainerLoadOptions` from your change, and the waiting on ops bit can be refactored into `Container.load` which would also let us isolate it's usage into the Loader (and we can deal with keeping it as a LoaderHeader option or whatever separately)